subsys/bluetooth/controller/multithreading_lock.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "multithreading_lock.h"
+
+static K_SEM_DEFINE(ble_controller_lock, 1, 1);
+
+int multithreading_lock_acquire(int32_t timeout)
+{
+	return k_sem_take(&ble_controller_lock,
```

Comment:
Hi @carlescufi , does Zephyr handle priority inversion in semaphores or only mutexes?

Reply:
Hi @ioannisg , @pizi-nordic maybe you can confirm? It does not look as it handles it.

Reply:
Personally I would prefer if mutex is used for locking.

Reply:
I have already commented it at the very beginning. I get a response that mutex cannot be used inside interrupts... well... ekhm...